src/bolt/bolt_api.c

```
@@ -75,52 +77,60 @@ static bool is_authenticated
 	bolt_read_string_size(&client->msg_buf.read, &len);
+	if(len >= sizeof(s)) return false;
 	bolt_read_string(&client->msg_buf.read, s);
 	if(strncmp(s, "basic", len) != 0) {
 		return false;
 	}
 
 	// check if the second key is principal
 	bolt_read_string_size(&client->msg_buf.read, &len);
+	if(len >= sizeof(s)) return false;
 	bolt_read_string(&client->msg_buf.read, s);
 	if(strncmp(s, "principal", len) != 0) {
```

Cppcheck (2.20.0)

[error] Common realloc mistake

(memleakOnRealloc)